src/test/workerRunner.ts

```diff
@@ -263,6 +269,37 @@ export class WorkerRunner extends EventEmitter {
         if (deadlineRunner)
           deadlineRunner.setDeadline(deadline());
       },
+      _log: data => this.emit('log', { testId, data }),
+      _wrapExpect: (matcherName: string, matcher: (...args: any[]) => any, receiver: any, args: any[]) => {
+        const seq = ++lastExpectSeq;
+        const expectBeginPayload: ExpectBeginPayload = { testId, seq, matcherName };
+        const expectEndPayload: ExpectEndPayload = { testId, seq };
+        this.emit('expectBegin', expectBeginPayload);
+        try {
+          const result = matcher.call(receiver, ...args);
+          expectEndPayload.result = { ...result };
+          if (receiver.isNot)
+            expectEndPayload.result.isNot = receiver.isNot;
+          if (result.pass === receiver.isNot && result.message)
+            expectEndPayload.result.message = result.message();
+          else
+            delete expectEndPayload.result.message;
+          if (result instanceof Promise) {
+            return result.catch(e => {
+              expectEndPayload.error = e.message;
+              throw e;
+            }).finally(() => {
+              this.emit('expectEnd', expectEndPayload);
+            });
+          }
+          return result;
+        } catch (e) {
+          expectEndPayload.error = e.message;
+          throw e;
+        } finally {
+          this.emit('expectEnd', expectEndPayload);
```

Comment:
It seems like we emit expectEnd twice for async expects, at line 300 and line 292.